cylc/flow/config.py

```diff
@@ -1617,7 +1617,7 @@ def generate_taskdefs(self, orig_expr, left_nodes, right, seq):
             # Record custom message outputs.
             for item in self.cfg['runtime'][name]['outputs'].items():
                 if item not in taskdef.outputs:
-                    taskdef.outputs.append(item)
+                    taskdef.outputs.add(item)
```

Comment:
@trwhitcomb - with reference to https://github.com/cylc/cylc-flow/pull/3259#discussion_r310849186 ... you could ditch the set membership test at 1619 and just add the items (that would definitely be more "Pythonic" whether or not there's a marginal speed advantage either way.).